### DISCUSSION TITLE
nested matching in where

### DISCUSSION THREAD

**Discussion by Rahmo** [prisma/prisma#5454] [Q&A] [answered]
HI All I have this heavy filter in GraphQL , I can't seems to get my head around this filter  I have this query with an input type
query  { items( filters: {search:"", region:"",market:"" , country:""} skip:0 take:50 )
.
.
I wanted something to apply for all. is my syntax is correct ?
"search" should apply to all of these columns 
then I have other parameters that I wanna check if they equals individually that value should show it.
	                                    
```
 OR [{
	AND: [
		{region: {contains: args.filters.search}},
		{region: {equals: args.filters.region}}
	]
    },
   {
	AND: [
		{market: {contains: args.filters.search}},
		{market: {equals: args.filters.market}}
	]
   },
  {
	AND: [
		{country: {contains: args.filters.search}},
		{country: {equals: args.filters.country}},
	]
  },
]
```

**Comment by Rahmo** [comment, marked as answer]
solved -  I got it  wrong 
what I really wanted is this
	AND: [
						{ region: { contains: args.filters.region } },
						{ market: { contains: args.filters.market } },
						{ country: { contains: args.filters.country } },
					],
				OR: [
						{ region: { contains: args.filters.search } },
						{ country: { contains: args.filters.search } },
						{ state: { contains: args.filters.search } },
						{ market: { contains: args.filters.search } },
						{ feeds: { contains: args.filters.search } },
						{ keys: { contains: args.filters.search } },
					],

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
